## Individual Project : Revised Version 
## By : Saloni Sharma 


## Graph 2 : 
#### (WORLD vs USA) Number of civilian owned guns compared to the Population :

In [1]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

import seaborn as sns
import pandas as pd
import numpy as np
%pylab inline

import sklearn as sk
import sklearn.tree as tree
from IPython.display import Image  

Populating the interactive namespace from numpy and matplotlib


In [2]:
#Importing the original dataset for firearms and homicides
import pandas as pd
df_firearm_2=pd.read_csv("firearm_2.csv")
len(df_firearm_2)

#Renaming "Country/Territory" column to "Country"
df_firearm_2.rename(columns={'Country/Territory':'Country'},inplace=True)
df_firearm_2.head(5)


185

Country ISO code               Source  % of homicides by firearm  \
0    Albania       AL                  CTS                       65.9   
1    Algeria       DZ                  CTS                        4.8   
2     Angola       AO                  NaN                        NaN   
3   Anguilla       AI              WHO-MDB                       24.0   
4  Argentina       AR  Ministry of Justice                       52.0   

   Number of homicides by firearm  Homicide by firearm rate per 100,000 pop  \
0                            56.0                                      1.76   
1                            20.0                                      0.06   
2                             NaN                                       NaN   
3                             1.0                                      7.14   
4                          1198.0                                      3.02   

   Rank by rate of ownership  Average firearms per 100 people  \
0                       70.0                              8.6   
1                       78.0                              7.6   
2                       34.0                             17.3   
3                        NaN                              NaN   
4                       62.0                             10.2   

   Average total all civilian firearms  
0                             270000.0  
1                            1900000.0  
2                            2800000.0  
3                                  NaN  
4                            3950000.0

In [3]:
#importing population data
import pandas as pd
df_population=pd.read_csv("population.csv")
len(df_population)
df_population.head(10)



254

Unnamed: 0  Unnamed: 1         Country    Population 
0         NaN         NaN           World  6,602,224,175
1         2.0         NaN           China  1,321,851,888
2         3.0         NaN           India  1,129,866,154
3         4.0         NaN  European Union    490,426,060
4         5.0         NaN   United States    301,139,947
5         6.0         NaN       Indonesia    234,693,997
6         7.0         NaN          Brazil    190,010,647
7         8.0         NaN        Pakistan    164,741,924
8         9.0         NaN      Bangladesh    150,448,339
9        10.0         NaN          Russia    141,377,752

In [4]:
#Removing the first 3 columns, and taking 1st row as data and not header
df_population.drop(['Unnamed: 0','Unnamed: 1'],axis=1,inplace=True)

In [5]:
df_population.head(5)

Country    Population 
0           World  6,602,224,175
1           China  1,321,851,888
2           India  1,129,866,154
3  European Union    490,426,060
4   United States    301,139,947

In [6]:
#Merging the 2 dataframes on "Country" column
df_merged=df_population.merge(df_firearm_2,on="Country")
df_merged.head(10)

Country    Population  ISO code              Source  \
0          China  1,321,851,888       CN                 NaN   
1          India  1,129,866,154       IN                 CTS   
2  United States    301,139,947       US                 CTS   
3      Indonesia    234,693,997       ID                 NaN   
4         Brazil    190,010,647       BR  Ministry of Health   
5       Pakistan    164,741,924       PK                 NaN   
6     Bangladesh    150,448,339       BD     National police   
7         Russia    141,377,752       RU                 NaN   
8        Nigeria    135,031,164       NG                 NaN   
9          Japan    127,433,494       JP             WHO-MDB   

   % of homicides by firearm  Number of homicides by firearm  \
0                        NaN                             NaN   
1                        7.6                          3093.0   
2                       60.0                          9146.0   
3                        NaN                             NaN   
4                       70.8                         34678.0   
5                        NaN                             NaN   
6                       43.6                          1456.0   
7                        NaN                             NaN   
8                        NaN                             NaN   
9                        1.8                            11.0   

   Homicide by firearm rate per 100,000 pop  Rank by rate of ownership  \
0                                       NaN                      102.0   
1                                      0.26                      110.0   
2                                      2.97                        1.0   
3                                       NaN                      169.0   
4                                     18.10                       75.0   
5                                       NaN                       57.0   
6                                      1.12                      169.0   
7                                       NaN                       68.0   
8                                       NaN                      133.0   
9                                      0.01                      164.0   

   Average firearms per 100 people  Average total all civilian firearms  
0                              4.9                           40000000.0  
1                              4.2                           46000000.0  
2                             88.8                          270000000.0  
3                              0.5                            1000000.0  
4                              8.0                           14840000.0  
5                             11.6                           18000000.0  
6                              0.5                             700000.0  
7                              8.9                           12750000.0  
8                              1.5                            2000000.0  
9                              0.6                             710000.0

In [8]:
#Dropping unnecessary columns
df_merged.drop(['ISO code','Source','% of homicides by firearm','Homicide by firearm rate per 100,000 pop',\
                'Rank by rate of ownership','Average firearms per 100 people','Number of homicides by firearm'],\
               axis=1,inplace=True)
df_merged.head(10)

Country    Population   Average total all civilian firearms
0          China  1,321,851,888                           40000000.0
1          India  1,129,866,154                           46000000.0
2  United States    301,139,947                          270000000.0
3      Indonesia    234,693,997                            1000000.0
4         Brazil    190,010,647                           14840000.0
5       Pakistan    164,741,924                           18000000.0
6     Bangladesh    150,448,339                             700000.0
7         Russia    141,377,752                           12750000.0
8        Nigeria    135,031,164                            2000000.0
9          Japan    127,433,494                             710000.0

In [9]:
#checking for null values 
df_merged.isnull().sum()

#removing the null values 
df_merged.dropna(inplace=True)

Country                                0
Population                             0
Average total all civilian firearms    9
dtype: int64

### To develop the Redesign visual 

In [15]:
len(df_merged)

164

164 countries are included in the dataset df_merged which I am going to use. Considering there are about 194 countries in the world, this dataset seems good enough. 

In [93]:
##try to find % difference of population and civilian owned guns of each country
df_merged2['Population ']=df_merged2['Population '].str.replace(',','').astype(float)
df_merged2['Average total all civilian firearms'] = df_merged2['Average total all civilian firearms'].astype(float)
df_merged2['Percent Difference']=(df_merged2['Average total all civilian firearms']/df_merged2['Population '])*100
df_merged2

df_merged3=df_merged2.sort_values(['Percent Difference'],ascending=[False]).head(10)
df_merged3

Country  Population   Average total all civilian firearms  \
2    United States  301139947.0                          270000000.0   
147          Qatar     907229.0                             520100.0   
46           Yemen   22230531.0                           11500000.0   
106        Finland    5238460.0                            2400000.0   
88     Switzerland    7554661.0                            3400000.0   
39            Iraq   27499638.0                            9750000.0   
148         Cyprus     788457.0                             275000.0   
124        Uruguay    3460607.0                            1100000.0   
82          Sweden    9031088.0                            2800000.0   
85         Austria    8199783.0                            2500000.0   

     Percent Difference  
2             89.659310  
147           57.328414  
46            51.730658  
106           45.814991  
88            45.005328  
39            35.455012  
148           34.878250  
124           31.786331  
82            31.004016  
85            30.488612

In [95]:
df_merged3.to_csv('percent_diff.csv')

Taking inspiration from the article itself which compares US with the rest of the world, I took US as a separate element and combined all the data of rest of the countries of the world. I combined both - the population data as well as the civilian owned guns data. 

In [118]:
df_merged4=df_merged2.copy()
df_merged4=df_merged4.rename(columns={'Population ':'Population',\
                                      'Average total all civilian firearms':'Civilian Firearms(Average)'})
us_pop=df_merged4.loc[2,'Population']
us_arms=df_merged4.loc[2,'Civilian Firearms(Average)']


pop_diff=df_merged4['Population'].sum()-df_merged4.loc[2,'Population']
arms_diff=df_merged4['Civilian Firearms(Average)'].sum()-df_merged4.loc[2,'Civilian Firearms(Average)']


World_and_usa=pd.DataFrame(columns=['Country','Population','Civilian Firearms(Average)'])
rows = [['USA',us_pop,us_arms],['World',pop_diff,arms_diff]]
for row in rows:
    World_and_usa.loc[len(World_and_usa)] = row
print(World_and_usa)
World_and_usa.to_csv('world_usa_deception.csv')





  Country    Population  Civilian Firearms(Average)
0     USA  3.011399e+08                 270000000.0
1   World  6.050804e+09                 365965400.0


### To develop the Deceptive Visual :

Considering that the dataset is already in descending order in terms of the 'Population', for the Deceptive Visualization, I would like to see the top 10 most populated countries with their average total all civilian firearms.

In [11]:
df_merged_pop=df_merged.head(10)
df_merged_pop

Country    Population   Average total all civilian firearms
0          China  1,321,851,888                           40000000.0
1          India  1,129,866,154                           46000000.0
2  United States    301,139,947                          270000000.0
3      Indonesia    234,693,997                            1000000.0
4         Brazil    190,010,647                           14840000.0
5       Pakistan    164,741,924                           18000000.0
6     Bangladesh    150,448,339                             700000.0
7         Russia    141,377,752                           12750000.0
8        Nigeria    135,031,164                            2000000.0
9          Japan    127,433,494                             710000.0

In [12]:
#Now the population based dataset is ready to be visualized. Exporting it to a csv file ..

df_merged_pop.to_csv("df_merged_pop_deception.csv")

I would like to sort the countries according to the 'Average total all civilian firearms.'

In [13]:
df_merged_fire=df_merged_pop.sort_values(['Average total all civilian firearms'],ascending=[False]).head(10)
df_merged_fire

Country    Population   Average total all civilian firearms
2  United States    301,139,947                          270000000.0
1          India  1,129,866,154                           46000000.0
0          China  1,321,851,888                           40000000.0
5       Pakistan    164,741,924                           18000000.0
4         Brazil    190,010,647                           14840000.0
7         Russia    141,377,752                           12750000.0
8        Nigeria    135,031,164                            2000000.0
3      Indonesia    234,693,997                            1000000.0
9          Japan    127,433,494                             710000.0
6     Bangladesh    150,448,339                             700000.0

In [14]:
#Exporting the file to csv 
df_merged_fire.to_csv("df_merged_fire_deception.csv")